handler/grpc.go

```diff
@@ -252,6 +266,59 @@ func (g grpcHandler) grpcToJSONRPC(srv interface{}, inStream grpc.ServerStream)
 	return nil
 }
 
+type wrapperServerStream struct {
+	sendHeaderCalled bool
+	stream           grpc.ServerStream
+	recvMessage      interface{}
+	sentMessage      interface{}
+}
+
+func (f *wrapperServerStream) SetTrailer(md metadata.MD) {
+	f.stream.SetTrailer(md)
+}
+
+func NewWrapperServerStream(stream grpc.ServerStream) (grpc.ServerStream, error) {
+	m := &codec.GrpcFrame{}
+	err := stream.RecvMsg(m)
```

Comment:
Why receiving message is needed here?

Reply:
the same message will be used to make a call to estimate price and also to the actual service invocation